core/store/data_input.hpp

```diff
@@ -109,6 +109,12 @@ struct data_input {
   data_input& operator++(int) noexcept { return *this; }
 };
 
+struct MemoryStats {
+  uint64_t* fd_count{};
+  uint64_t* mmaped_memory{};
+  uint64_t* pinned_memory{};
```

Comment:
TBH I don't see any value in having stats as pointers.

Reply:
It should be pointer or reference, reference is less flexible

Reply:
The plan is to "opt-out" collecting some data, if necessary, by providing nullptr.